cmd/synthetic-monitoring-agent/http.go

```diff
@@ -112,3 +117,21 @@ func defaultHandler() http.Handler {
 		fmt.Fprintln(w, "hello, world!")
 	})
 }
+
+// readyHandler is a readiness probe.
+func readyHandler(isConnected, isReady *atomic.Value) http.HandlerFunc {
```

Comment:
I don't follow why you need two variables now.

If the intention is to set isReady to true and never back to false, it's enough to do something like `func (v bool) { if v { isReady.Store(v) } }` in the isConnected function.